core/storage/feedback/gae_models.py

```diff
@@ -304,6 +304,32 @@ def get_messages(cls, exploration_id, thread_id):
             cls.thread_id == full_thread_id).fetch(feconf.DEFAULT_QUERY_LIMIT)
 
     @classmethod
+    def get_last_two_messages_of_threads(cls, exploration_ids, thread_ids):
+        """Returns a list of the last two messages of the given threads.
+
+        Args:
+            exploration_id: str. ID of the exploration the thread
+                belongs to.
+            thread_id: str. ID of the thread.
+
+        Returns:
+            list(int). A list of message ids present in the given thread, having
+                a maximum length of feconf.DEFAULT_QUERY_LIMIT.
+        """
+        full_thread_ids = [FeedbackThreadModel.generate_full_thread_id(
+            exploration_id, thread_id) for exploration_id, thread_id in zip(
+                exploration_ids, thread_ids)]
+
+        messages = []
+
+        for full_thread_id in full_thread_ids:
```

Comment:
We can limit this to about 100 feedback threads. If we approximate that each thread will have about 5 messages (because first, it would have some feedback by the learner, then the creator will ask for some clarification, the learner clears it up and then the creator fixes the issue) then it takes about 5 seconds to fetch 100 threads having 5 messages. I have checked this in a test that I have included in this PR and I have kept the threshold at 8 seconds for safety purposes.

So would a limit of 100 do? We would choose the 100 based on last updated.